lib/wasix/src/syscalls/wasix/sock_recv.rs

```diff
@@ -113,8 +130,26 @@ fn sock_recv_internal<M: MemorySize>(
                     .access()
                     .map_err(mem_error_to_wasi)?;
 
+                let nonblocking = waker.is_none() && fd.flags.contains(Fdflags::NONBLOCK);
+                let timeout = if waker.is_none() {
+                    Some(
+                        socket
+                            .opt_time(TimeType::ReadTimeout)
+                            .ok()
+                            .flatten()
+                            .unwrap_or(Duration::from_secs(30)),
+                    )
```

Comment:
again, magic timeout.... that seems quite suboptimal?

Reply:
its a default, not having a default means sockets read forever, which is not good as a default